### PR TITLE
Clu test project and updates to clu-getstart.md

### PR DESCRIPTION
**Pull request by BurtBiel**
Clu test project and updates to clu-getstart.md

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@BurtBiel__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (burtbiel). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/CLU/Microsoft.CLU.Test/StringGenerator.cs [hunk 1]
```diff
@@ -1,35 +0,0 @@
-﻿using System;
```
Comment (BurtBiel) [inline src/CLU/Microsoft.CLU.Test/StringGenerator.cs]: This file was unused, so deleting it for now

File: src/CLU/Microsoft.CLU/CommandBinder/CmdletBinderAndCommand.cs [hunk 2]
```diff
@@ -177,7 +177,7 @@ public IEnumerable<string> ListCommands(string[] args)
 #if PSCMDLET_HELP
             return CmdletHelp.Generate(parser.FormatParameterName, _discriminatorBinder.Modules, args, prefix);
 #else
-            var commands = Microsoft.CLU.Help.CommandDispatchHelper.CompleteCommands(CLUEnvironment.GetPackagesRootPath(), args).ToArray();
+            var commands = Microsoft.CLU.Help.CommandDispatchHelper.CompleteCommands(new HelpPackageFinder(CLUEnvironment.GetPackagesRootPath()), args).ToArray();
             if (commands.Length > 0)
```
Comment (BurtBiel) [inline src/CLU/Microsoft.CLU/CommandBinder/CmdletBinderAndCommand.cs]: Factored out the package path into HelpPackageFinder so we can create our own packages in the unit tests

File: src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs [hunk 3]
```diff
@@ -95,28 +95,14 @@ public static IEnumerable<CommandInfo> FindCommandMatches(string pkgRoot, string
...
-            foreach (var pkgPath in rootInfo.EnumerateDirectories())
-            {
-                foreach (var versionPath in pkgPath.EnumerateDirectories())
-                {
-                    yield return new PkgInfo(pkgPath.Name, versionPath.Name, pkgPath.FullName);
-                }
-            }
-
-        }
-
```
Comment (BurtBiel) [inline src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs]: Factored out

**Comment by BurtBiel** [comment]
Tests pass in VS and with dnx test

File: clu-getstart.md [hunk 4]
```diff
@@ -25,7 +25,7 @@ CLUPackages require some additional files to direct generation of indexing, and
     | ------------- |:-------------:|
     | CommandAssemblies      | File name of cmdlets assembly(ies) |
     | NounPrefix      | ‘AzureRm’  The part of the cmdlet noun to remove in clu commands|
-    | NounFirst       | if true, the verb comes at the end of the command (e.g. az resource get)|
+    | NounFirst       | if true, the verb comes at the end of the command (e.g. az resourcemanager resource ls)|
```
Comment (markcowl) [inline clu-getstart.md]: @BurtBiel this should be az resource ls

File: src/CLU/Microsoft.CLU/CommandBinder/CmdletBinderAndCommand.cs [hunk 5]
```diff
@@ -177,7 +177,7 @@ public IEnumerable<string> ListCommands(string[] args)
 #if PSCMDLET_HELP
             return CmdletHelp.Generate(parser.FormatParameterName, _discriminatorBinder.Modules, args, prefix);
 #else
-            var commands = Microsoft.CLU.Help.CommandDispatchHelper.CompleteCommands(CLUEnvironment.GetPackagesRootPath(), args).ToArray();
+            var commands = Microsoft.CLU.Help.CommandDispatchHelper.CompleteCommands(new HelpPackageFinder(CLUEnvironment.GetPackagesRootPath()), args).ToArray();
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/CommandBinder/CmdletBinderAndCommand.cs]: @BurtBiel Break this line, please

File: src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs [hunk 6]
```diff
@@ -185,7 +186,7 @@ public string FormatParameterName(string name, string type, bool isMandatory, bo
 
         private void PresentCommandCompletion(string[] arguments)
         {
-            foreach (var cmd in Help.CommandDispatchHelper.CompleteCommands(CLUEnvironment.GetPackagesRootPath(), 
+            foreach (var cmd in Help.CommandDispatchHelper.CompleteCommands(new HelpPackageFinder(CLUEnvironment.GetPackagesRootPath()),
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs]: @BurtBiel same

File: src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs [hunk 7]
```diff
@@ -65,14 +65,14 @@ public static HelpInfo FindBestHelp(string pkgRoot, string[] args)
...
             };
 
-            return FindHelpMatches(pkgRoot, args, matcher).OrderByDescending((hi) => 
+            return FindHelpMatches(helpPackageFinder, args, matcher).OrderByDescending((hi) => 
                    CommandDispatchHelper.MatchScore(hi.Discriminators, semiColonSeparatedArgs)).ThenBy((hi) => 
                    hi.Discriminators).FirstOrDefault();
         }
 
-        public static IEnumerable<HelpInfo> FindHelpMatches(string pkgRoot, string[] args, Func<string, bool> matchFunc)
+        public static IEnumerable<HelpInfo> FindHelpMatches(IHelpPackageFinder helpPackageFinder, string[] args, Func<string, bool> matchFunc)
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs]: breaks

File: src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs [hunk 8]
```diff
@@ -82,9 +82,9 @@ public static IEnumerable<HelpInfo> FindHelpMatches(string pkgRoot, string[] arg
             }
         }
 
-        public static IEnumerable<CommandInfo> FindCommandMatches(string pkgRoot, string[] args, Func<string, bool> matchFunc)
+        public static IEnumerable<CommandInfo> FindCommandMatches(IHelpPackageFinder helpPackageFinder, string[] args, Func<string, bool> matchFunc)
```
Comment (markcowl) [inline src/CLU/Microsoft.CLU/Help/CommandDispatchHelper.cs]: breaks

**Comment by BurtBiel** [comment]
Addressed core review feedback

**Comment by BurtBiel** [comment]
It looks like one of our dependencies was removed, so I updated the Microsoft.CLU.Run 
"NuGet.Commands": "3.3.0-local-*",  -> project.json: "NuGet.Commands": "3.3.0", then the build passed

**Comment by BurtBiel** [comment]
@markcowl addressed code review feedback and updated a nugget package resource to get the build passing

**Pull request merged**